File: sequences/basic.py
def naturals():
    for i in nonnegative_integers():
        yield i + 1


def nonnegative_integers():
    i = 0
    while True:
        yield i
        i = i + 1


if __name__ == '__main__':
    for i, x in zip(range(1, 100), naturals()):
        print(i, x)

File: sequences/src/Basic.hs
module Basic where

_sequence :: Integer -> [Integer]
_sequence start = start : map (+ 1) (_sequence start)

naturals :: [Integer]
naturals = _sequence 1

a000027 :: [Integer]
a000027 = naturals

up_to :: Int -> [Integer]
up_to = (flip take) naturals

nonnegative_integers :: [Integer]
nonnegative_integers = _sequence 0

numbers :: [Integer]
numbers = nonnegative_integers

a001477 :: [Integer]
a001477 = numbers

